account_reconciliation_widget/models/reconciliation_widget.py

```diff
@@ -1050,7 +1050,7 @@ def _get_statement_line(self, st_line):
             "date": format_date(self.env, st_line.date),
             "amount": amount,
             "amount_str": amount_str,  # Amount in the statement line currency
-            "currency_id": st_line_currency.id,  # Currency according to statement line
+            "currency_id": (st_line.currency_id or statement_currency).id,
```

Comment:
Shouldn't this be directly `statement_currency.id`? @singwangho why did you change this?

Reply:
@pedrobaeza it is used to match currency of `amount_str` (a 1041 line).

Reply:
Hi, I can replicate @gdrius issue too.

I changed it to revert this back to "#437 Fix foreign_currency_id vs currency_id" which has following code
`"currency_id": st_line.foreign_currency_id.id or statement_currency.id,`

But I think @gdrius is correct that `currency_id` should match `amount_str`.